migration/incremental/015-tracing-compression.sql

```diff
@@ -261,82 +261,101 @@ GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE _ps_trace.link TO prom_writer;
+            chunk_time_interval=>'1 hours'::interval,
+            create_default_indexes=>false
+        );
+        PERFORM public.create_distributed_hypertable(
+            '_ps_trace.link'::regclass,
+            'span_start_time'::name,
+            partitioning_column=>'trace_id'::name,
+            number_partitions=>1::int,
+            chunk_time_interval=>'1 hours'::interval,
+            create_default_indexes=>false
+        );
+    ELSE -- not multinode
+        PERFORM public.create_hypertable(
+            '_ps_trace.span'::regclass,
+            'start_time'::name,
+            partitioning_column=>'trace_id'::name,
+            number_partitions=>1::int,
+            chunk_time_interval=>'1 hours'::interval,
+            create_default_indexes=>false
+        );
+        PERFORM public.create_hypertable(
+            '_ps_trace.event'::regclass,
+            'time'::name,
+            partitioning_column=>'trace_id'::name,
+            number_partitions=>1::int,
+            chunk_time_interval=>'1 hours'::interval,
+            create_default_indexes=>false
+        );
+        PERFORM public.create_hypertable(
+            '_ps_trace.link'::regclass,
+            'span_start_time'::name,
+            partitioning_column=>'trace_id'::name,
+            number_partitions=>1::int,
+            chunk_time_interval=>'1 hours'::interval,
+            create_default_indexes=>false
+        );
+    END IF;
+
+    _is_compression_available = NOT (current_setting('timescaledb.license') = 'apache');
+    IF _is_compression_available THEN
+        -- turn on compression
+        ALTER TABLE _ps_trace.span SET (timescaledb.compress, timescaledb.compress_orderby='trace_id,span_id,start_time');
+        ALTER TABLE _ps_trace.event SET (timescaledb.compress, timescaledb.compress_orderby='trace_id,span_id,time');
+        ALTER TABLE _ps_trace.link SET (timescaledb.compress, timescaledb.compress_orderby='trace_id,span_id,span_start_time');
+
+        PERFORM public.add_compression_policy('_ps_trace.span', INTERVAL '1 hours');
+        PERFORM public.add_compression_policy('_ps_trace.event', INTERVAL '1 hours');
+        PERFORM public.add_compression_policy('_ps_trace.link', INTERVAL '1 hours');
+
+        -- we do not want the compressed hypertables to be associated with the extension
+        -- we want both the table definitions and data to be dumped by pg_dump
+        -- we cannot call create_hypertable during the restore, so we want the definitions in the dump
+        FOR _rec IN
+        (
+            SELECT c.schema_name, c.table_name
+            FROM _timescaledb_catalog.hypertable h
+            INNER JOIN _timescaledb_catalog.hypertable c
+            ON (h.compressed_hypertable_id = c.id)
+            WHERE h.schema_name = '_ps_trace'
+            AND h.table_name IN ('span', 'link', 'event')
+        )
+        LOOP
+            RAISE NOTICE 'removing %.% from extension', _rec.schema_name, _rec.table_name;
```

Comment:
do we want this notice? why?

Reply:
nope. leftover debugging cruft. removed in [2088920](https://github.com/timescale/promscale_extension/pull/185/commits/2088920c124ee87330170ec036092ace51fd4090)